pkg/stub/handler.go

```diff
@@ -201,32 +200,17 @@ func (h *Handler) prepSamplesWatchEvent(kind, name string, annotations map[strin
 			logrus.Printf("watch stream event %s not part of operators inventory", name)
 			// we now have cases where sample providers are deleting entire imagestreams;
 			// let's make sure there are no stale entries with inprogress / importerror
-			processing := util.Condition(cfg, v1.ImageChangesInProgress)
-			needToUpdate := util.NameInReason(cfg, processing.Reason, name)
-			if needToUpdate {
-				now := kapis.Now()
-				processing.Reason = util.ClearNameInReason(cfg, processing.Reason, name)
-				logrus.Debugf("processing reason now %s", processing.Reason)
-				if len(strings.TrimSpace(processing.Reason)) == 0 {
-					logrus.Println("The last in progress imagestream has completed (removed imagestream check)")
-					processing.Status = corev1.ConditionFalse
-					processing.Reason = ""
-					processing.LastTransitionTime = now
-				}
-				processing.LastUpdateTime = now
-				util.ConditionUpdate(cfg, processing)
-			}
-			importErrors := util.Condition(cfg, v1.ImportImageErrorsExist)
-			needToUpdate2 := util.NameInReason(cfg, importErrors.Reason, name)
-			if needToUpdate2 {
-				h.mapsMutex.Lock()
-				defer h.mapsMutex.Unlock()
-				h.clearStreamFromImportError(name, importErrors, cfg)
+			_, err := h.configmapclientwrapper.Get(name)
+			if err != nil && !kerrors.IsNotFound(err) {
+				// CRUD get err should be intermittent ... just try again
+				return nil, "", false, err
 			}
-			if needToUpdate || needToUpdate2 {
-				return cfg, "", false, true, nil
+			if err != nil {
+				// already not found
+				return nil, "", false, nil
```

Comment:
nit - I'd clarify this a bit to return on the expected "not found" error:

```
if err != nil && kerrors.IsNotFound(err) {
  // ConfigMap should only exist if the imagestream has an error
  return nil, "", false, nil
}
if err != nil {
  // GET errors indicate a potential issue with the apiserver, return error and try again
  return nil, "", false, err
}
```

Reply:
will push momentarily